lib/core/versions/latest/Operation.ts

```diff
@@ -282,6 +272,16 @@ export default class Operation {
     const decodedPayloadJson = Encoder.decodeAsString(operation.payload);
     const decodedPayload = JSON.parse(decodedPayloadJson);
 
+    // Get the operation type.
+    const operationType = decodedPayload.type;
+
+    // 'type' property must be one of the allowed strings.
+    const allowedOperations = new Set([OperationType.Create, OperationType.Update, OperationType.Delete, OperationType.Recover]);
```

Comment:
nit: I'm pretty sure this can just be `new Set(Object.values(OperationType))` this way we don't need to hard code the values and add them in the future.

Reply:
This is great suggestion. Took it already!